### DISCUSSION TITLE
Polynomial order of external mesh

### DISCUSSION THREAD

**Discussion by aiskhak** [idaholab/moose#19017] [Q&A Meshing] [answered]
Hi guys,

I am using a MOOSE-based application (SAM) for 3D fluid dynamics simulations. I have a relatively complex geometry and my mesh is built in ANSYS (scdoc) format. To get .exo format I do the following in ANSYS ICEM:

1) import model
2) merge all blocks
3) output mesh

by default, it is HEX8/QUAD4 mesh, which is suitable for FIRST polynomial order (LAGRANGE). This works fine.

I want to do simulations for SECOND order. If I specify so, I am getting the error:

Error: Exodus file says 20 nodes per Elem, but Elem type HEX8 has 8 nodes.
[0] ../src/mesh/exodusII_io.C, line 253, compiled Sep 20 2021 at 12:34:20

Then I tried to "Create node on interface" in ICEM and make the mesh HEX20/QUAD8. However, I am getting the same error in my simulation.

What is the problem? Thanks!

**Comment by GiudGiud** [comment, marked as answer]
I dont know about a workflow to add the necessary nodes for second order. I suppose setting `second_order = true` in the [Mesh] block does not work?
@lindsayad do you know more about this?

**Reply by aiskhak** [reply]
Thank you for your answer. No, `second_order = true` does not work.

**Reply by lindsayad** [reply]
Using the same file you use for your HEX8 mesh and then specifying `second_order = true` in the mesh block *should* work. You say it doesn't work...do you get an error? Can you share your mesh block please?

**Reply by aiskhak** [reply]
No, I am not getting errors. The simulation just does not start. It is sort of frozen in the very beginning.


[Mesh]
  type = FileMesh
  file = hirjet_1st.exo
  second_order = true
[]

**Reply by GiudGiud** [reply]
Weird. it might just be working on the mesh.
Is your moose install recent? The current version should tell you what it s doing when it takes too long

**Reply by aiskhak** [reply]
yes, I installed MOOSE yesterday.

**Reply by aiskhak** [reply]
Is there any other way to save a moose-compatible mesh from .scdoc format?

**Reply by GiudGiud** [reply]
How big is that mesh?

**Reply by aiskhak** [reply]
It is a 3D case:
--- Mesh Info ---
Element types :
        HEXA_8 : 42756
        QUAD_4 : 24016
Element parts :
        ALL : 59952
        INLET : 180
        OUTLET : 180
        WALL : 6460
Total elements : 66772
Total nodes : 47044
Min : -0.2027 -0.3548 -0.3048
Max : 0.5672 0.2413 0.3048

**Reply by GiudGiud** [reply]
I dont know much about scdoc. 
I wonder if having the Quads in the 3D mesh is the issue for loading it. Can you share that mesh with us?

**Reply by aiskhak** [reply]
I am sorry. Everything works fine. The problem was in the update of the cluster, so the output file was written in a different place. Sorry about that. BIG-BIG thanks for your help!

**Reply by lindsayad** [reply]
Glad it's working!
